quesma/clickhouse/clickhouse.go

```diff
@@ -482,8 +496,38 @@ func generateNonSchemaFieldsString(attrsMap map[string][]interface{}) (string, e
 	return nonSchemaStr, nil
 }
 
+// This function implements heuristic for deciding if we should add new columns
+func (lm *LogManager) shouldAlterColumns(table *Table, attrsMap map[string][]interface{}) (bool, []int) {
+	attrKeys := getAttributesByArrayName(AttributesKeyColumn, attrsMap)
+	alterColumnIndexes := make([]int, 0)
+	if len(table.Cols) < maxColumns {
```

Comment:
This should be configurable, but feel free to postpone after this PR.

Though let's rename this constant `alwaysAddColumnLimit`.

Reply:
ok